jobs/gtfs-rt-parser-v2/gtfs_rt_parser.py

```diff
@@ -0,0 +1,740 @@
+            except DecodeError as e:
+                if verbose:
+                    log(
+                        f"WARNING: DecodeError for {str(extract.path)}",
+                        fg=typer.colors.YELLOW,
+                        pbar=pbar,
+                    )
+                outcomes.append(
+                    RTFileProcessingOutcome(
+                        step="parse",
+                        success=False,
+                        exception=e,
+                        extract=extract,
+                    )
+                )
+                continue
+
+            if not parsed or "entity" not in parsed:
+                msg = f"WARNING: no parsed entity found in {str(extract.path)}"
+                if verbose:
+                    log(
+                        msg,
+                        fg=typer.colors.YELLOW,
+                        pbar=pbar,
+                    )
+                outcomes.append(
+                    RTFileProcessingOutcome(
+                        step="parse",
+                        success=False,
+                        exception=ValueError(msg),
+                        extract=extract,
+                    )
+                )
+                continue
+
+            for record in parsed["entity"]:
+                gzipfile.write(
+                    (
+                        json.dumps(
+                            {
+                                "header": parsed["header"],
+                                # back and forth so we use pydantic serialization
+                                "metadata": serialize_pydantic_model_for_bigquery(
+                                    extract
+                                ),
+                                **copy.deepcopy(record),
+                            }
+                        )
+                        + "\n"
+                    ).encode("utf-8")
+                )
+                written += 1
+            outcomes.append(
+                RTFileProcessingOutcome(
+                    step="parse",
+                    success=True,
+                    extract=extract,
+                )
+            )
+            del parsed
+
+    if written:
+        log(
+            f"writing {written} lines to {hour.path}",
+            pbar=pbar,
+        )
+        put_with_retry(fs, gzip_fname, hour.path)
+    else:
+        log(
+            f"WARNING: no records at all for {hour.path}",
+            fg=typer.colors.YELLOW,
+            pbar=pbar,
+        )
+
+    return outcomes
+
+
+# Originally this whole function was retried, but tmpdir flakiness will throw
+# exceptions in backoff's context, which ruins things
+def parse_and_validate(
+    hour: RTHourlyAggregation,
+    jar_path: Path,
+    tmp_dir: str,
+    verbose: bool = False,
+    pbar=None,
+) -> List[RTFileProcessingOutcome]:
+    fs = get_fs()
+    dst_path_rt = f"{tmp_dir}/rt_{hashlib.md5(hour.name.encode('utf-8')).hexdigest()}/"
```

Comment:
why do you need this set again here? can it not be accessed from the `RTHourlyAggregation`?

Reply:
It's slightly different (this is a unique folder for the raw RT files, not the unique filename of a combined hour) but yes I think I can move this into `RTHourlyAggregation`